pywhat/printer.py

```diff
@@ -26,38 +26,47 @@ def pretty_print(self, text: dict):
             table.add_column("Matched Text", overflow="fold")
             table.add_column("Identified as", overflow="fold")
             table.add_column("Description")
-            for i in text["Regexes"]:
-                matched = i["Matched"]
-                name = i["Regex Pattern"]["Name"]
-                description = None
+            table.add_column("Filename", overflow="fold")
 
-                if "URL" in i["Regex Pattern"]:
-                    description = (
-                        "Click here to analyse in the browser "
-                        + i["Regex Pattern"]["URL"]
-                        + matched.replace(" ", "")
-                    )
+            for key, value in text["Regexes"].items():
+                for i in value:
+                    matched = i["Matched"]
+                    name = i["Regex Pattern"]["Name"]
+                    description = None
+                    filename = key
 
-                if i["Regex Pattern"]["Description"]:
-                    if description:
+                    if "URL" in i["Regex Pattern"]:
                         description = (
-                            description + "\n" + i["Regex Pattern"]["Description"]
+                            "Click here to analyse in the browser "
+                            + i["Regex Pattern"]["URL"]
+                            + matched.replace(" ", "")
+                        )
+
+                    if i["Regex Pattern"]["Description"]:
+                        if description:
+                            description = (
+                                description + "\n" + i["Regex Pattern"]["Description"]
+                            )
+                        else:
+                            description = i["Regex Pattern"]["Description"]
+
+                    if description:
+                        table.add_row(
+                            matched,
+                            name,
+                            description,
+                            filename

```

Comment:
Do we add a row on API input? Does that cause a problem?

Reply:
Yes it is also added with the API and it works the same without any problems - if there is a file, this is a filename; if not, it is a string `None`

Reply:
Nevermind. Only the `identifier` is executed in the API mode without `printer`, so there are no tables or rows in the API mode :)

But the API results are a little different as there is a `filename` added to each `Regexes` and `File Signatures`. It now also accepts file paths, so if input is a folder this will be quite useful. With normal text as input this is `text`.  An example:
```
{
    'File Signatures': {
        'text': None
        },

    'Regexes': {
        'text': [
            {
                'Matched': 'DANHz6EQVoWyZ9rER56DwTXHWUxfkv9k2o',
                'Regex Pattern': {
                    'Name': 'Dogecoin (DOGE) Wallet Address',
                    'Regex': '^D{1}[5-9A-HJ-NP-U]{1}[1-9A-HJ-NP-Za-km-z]{32}$',
                    'plural_name': False,
                    'Description': None,
                    'URL': 'https://dogechain.info/address/',
                    'Rarity': 0.5,
                    'Tags': [
                        'Cryptocurrency',
                        'Finance'
                        ]
                    }
                }
        ]
    }
}
```